### PR TITLE
docs: update changelog for v2.4.0

### PR DESCRIPTION
**Pull request by perk-sumo**
##### Description

Fill in your description here.

---

##### Checklist

Remove items which don't apply to your PR.

- [ ] Changelog updated

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,11 +13,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
-[Unreleased]: https://github.com/SumoLogic/sumologic-kubernetes-collection/compare/v2.4.0-rc.0...main
+[Unreleased]: https://github.com/SumoLogic/sumologic-kubernetes-collection/compare/v2.4.0...main
 
-## [v2.4.0_rc.0][v2_4.0_rc.0]
+## [v2.4.0][v2_4.0]
```
Comment (pmalek-sumo) [inline CHANGELOG.md:18]: The link in line 43 needs fixing :)

Comment (perk-sumo) [reply CHANGELOG.md:18]: Good catch! I've changed that but didn't save the file 🤦

Comment (perk-sumo) [reply CHANGELOG.md:18]: It's fixed now.

Comment (pmalek-sumo) [inline CHANGELOG.md:18]: `v2_4.0` looks strange 🤷

**Review by pmalek-sumo — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (perk-sumo) [reply CHANGELOG.md]: Fixed, thank you!

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**